src/sentry/monitors/models.py

```diff
@@ -522,7 +527,8 @@ def mark_failed(
         elif reason == MonitorFailure.DURATION:
             new_status = MonitorStatus.TIMEOUT
 
-        next_checkin = self.monitor.get_next_scheduled_checkin(next_checkin_base)
+        next_checkin = self.monitor.get_next_expected_checkin(next_checkin_base)
+        next_checkin_latest = self.monitor.get_next_expected_checkin_latest(next_checkin_base)
```

Comment:
the reason i added that helper function was to not have to call `get_next_expected_checkin` twice, but not sure if there's a cleaner way to do it

Reply:
ohhhh, doy. Makes sense

Reply:
Are you worried about performance here?

Reply:
I think I'm not too worried about the performance of this, let's keep it like this actually, it's a little more obvious what's going on I think

Reply:
i'm not worried either but just felt off to do it twice in a row